services/datamanager/model/model.go

```diff
@@ -0,0 +1,324 @@
+)
+
+const (
+	appAddress   = "app.viam.com:443"
+	zipExtension = ".zip"
+)
+
+// ViamModelDotDir is the default model directory a model will deployed into,
+// unless otherwise specified.
+var ViamModelDotDir = filepath.Join(os.Getenv("HOME"), ".viam", "models")
+
+// Model describes a model we want to download to the robot.
+type Model struct {
+	Name        string `json:"source_model_name"`
+	Destination string `json:"destination"`
+}
+
+// HTTPClient allows us to mock a connection.
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Manager is responsible for deploying model files.
+type Manager interface {
+	DownloadModels(cfg *config.Config, modelsToDeploy []*Model, errorChannel chan error) chan error
+	Close()
+}
+
+// modelManager is responsible for uploading files in captureDir to the cloud.
+type modelManager struct {
+	partID     string
+	conn       rpc.ClientConn
+	client     v1.ModelServiceClient
+	logger     golog.Logger
+	cancelFunc func()
+	httpClient httpClient
+}
+
+// ManagerConstructor is a function for building a Manager.
+type ManagerConstructor func(logger golog.Logger, cfg *config.Config) (Manager, error)
+
+// NewDefaultManager returns the default Manager that syncs data to app.viam.com.
+func NewDefaultManager(logger golog.Logger, cfg *config.Config) (Manager, error) {
+	tlsConfig := config.NewTLSConfig(cfg).Config
+	cloudConfig := cfg.Cloud
+	rpcOpts := []rpc.DialOption{
+		rpc.WithTLSConfig(tlsConfig),
+		rpc.WithEntityCredentials(
+			cloudConfig.ID,
+			rpc.Credentials{
+				Type:    rdkutils.CredentialsTypeRobotSecret,
+				Payload: cloudConfig.Secret,
+			}),
+	}
+
+	conn, err := NewConnection(logger, appAddress, rpcOpts)
+	if err != nil {
+		return nil, err
+	}
+	client := NewClient(conn)
+	return NewManager(logger, cfg.Cloud.ID, client, conn, &http.Client{})
+}
+
+// NewManager returns a new model Manager.
+func NewManager(logger golog.Logger, partID string, client v1.ModelServiceClient,
+	conn rpc.ClientConn, httpClient httpClient,
+) (Manager, error) {
+	_, cancelFunc := context.WithCancel(context.Background())
+	ret := modelManager{
+		conn:       conn,
+		client:     client,
+		logger:     logger,
+		cancelFunc: cancelFunc,
+		partID:     partID,
+		httpClient: httpClient,
+	}
+	return &ret, nil
+}
+
+func (m *modelManager) deploy(ctx context.Context, req *v1.DeployRequest) (*v1.DeployResponse, error) {
+	resp, err := m.client.Deploy(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// Close all resources (goroutines) associated with modelManger.
+func (m *modelManager) Close() {
+	m.cancelFunc()
+	if m.conn != nil {
+		if err := m.conn.Close(); err != nil {
+			m.logger.Errorw("error closing model deploy server connection", "error", err)
+		}
+	}
+}
+
+// DownloadModels handles deploying models into their specified destination.
+func (m *modelManager) DownloadModels(cfg *config.Config, modelsToDeploy []*Model, errorChannel chan error) chan error {
```

Comment:
I may have missed previous context, but I think this should not return anything. The error channel passed in is to ensure that when this is concurrent, the caller actually receives the error and can act on it.

Reply:
gotcha, we now return nothing